Repository/Api/EditableRepository.php

```diff
@@ -39,4 +39,13 @@
      * @throws UnsupportedLanguageException If the language is not supported.
      */
     public function move($sourceQuery, $targetPath, $language = 'glob');
+
+    /**
+     * Change the position of a node relative to its siblings.
+     *
+     * @param string $sourcePath
+     * @param int $position
+     */
+    public function reposition($sourcePath, $position);
```

Comment:
with a bit of thinking, i see why you chose the name "reposition". but on first glance, i was confused how this relates to "move". at least to my non-native english, i feel "position" could also be "the position within the tree".

does puli know this concept? if puli calls it reposition, we should stick with the puli name here. if not, i would prefer "reorder" as it seems clearer to me.

regarding index or name: jcr went for name, not sure what their reasoning is. in terms of phpcr, name seems better to me therefore. its closer to how a human thinks when moving the node (i want this positioned after that) probably (except when moving to end or beginning). both have issues with concurrency, i see no plus or minus there.

if this api is for other things than PHPCR as well, numeric position is probably more universal than positioning after a name.